data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/threshold/ThresholdCheck.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.threshold;
+
+import org.opensearch.dataprepper.plugins.sink.buffer.Buffer;
+
+/**
+ * ThresholdCheck receives paramaters for which to reference the
+ * limits of a buffer and CloudWatchLogsClient before making a
+ * PutLogEvent request to AWS.
+ */
+public class ThresholdCheck {
+    private final int batchSize;
+    private final int maxEventSize;
+    private final int maxRequestSize;
+    private final int logSendInterval;
+
+    ThresholdCheck (int batchSize, int maxEventSize, int maxRequestSize, int logSendInterval) {
```

Comment:
All the parameters should be final here and the below methods

Reply:
I have made the parameters final for the next revision.